### ISSUE TITLE
[feature] Auto generate ssh keys for ec2 driver.

### ISSUE THREAD

**Issue by mhristof** [metacloud/molecule#1139]
# Issue Type

- Feature request

# Desired Behaviour

When trying to use molecule to test ansible roles with the ec2 driver, i have to provide an ssh key (that might change depending on the user that is running molecule (travis, me, etc)).

I'd suggest that a default value is provided, thats gets the key either from the default location `~/.ssh/id_rsa.pub`, or automatically generates one that is destroyed at the end of the run.

Basically, im thinking the same behaviour as [Hashicorps Packer](https://www.packer.io/docs/builders/amazon-ebs.html#ssh_keypair_name).

Let me know what you think.

**Comment by retr0h** [comment]
Molecule simply provides templates for use with various cloud providers.  They are provided so that the user can do with them what they choose.  If you wish to provide ssh keys from a location, feel free to modify the templates to do so.
